metadata-io/src/main/java/com/linkedin/metadata/search/elasticsearch/query/request/SearchQueryBuilder.java

```diff
@@ -72,22 +81,59 @@ private static QueryBuilder buildInternalQuery(@Nonnull EntitySpec entitySpec, @
     return finalQuery;
   }
 
-  private static FunctionScoreQueryBuilder.FilterFunctionBuilder[] buildFilterFunctions(
-      @Nonnull EntitySpec entitySpec) {
-    return entitySpec.getSearchableFieldSpecs()
+  private static FunctionScoreQueryBuilder.FilterFunctionBuilder[] buildScoreFunctions(@Nonnull EntitySpec entitySpec) {
+    List<FunctionScoreQueryBuilder.FilterFunctionBuilder> finalScoreFunctions = new ArrayList<>();
+    // Add a default weight of 1.0 to make sure the score function is larger than 1
```

Comment:
why tho?

Reply:
This is in case all features are missing, in which case the functino score will be 0. Since we multiply functino score with the query score, if the function score is 0, it will make all the scores the same regardless of how well the query matches the field values

Reply:
Okay nice!